### ISSUE TITLE
Build Error with AGP 8.0+: Namespace not specified

### ISSUE THREAD

**Issue by aymanalhalali** [LeGoffMael/insta_assets_picker#38]
Hello,

I've recently updated my project's Flutter version and the Android Gradle Plugin to the latest version, and after doing so, I encountered a build issue with the `insta_assets_picker` package. The error occurs when trying to configure the project:

A problem occurred configuring project ':insta_assets_picker'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

It seems that the newer version of AGP requires a namespace declaration in the module's `build.gradle` file. Here is the environment in which the error occurred:

- Flutter Version: 3.16.5
- Dart Version: 3.2.3
- Android Gradle Plugin Version: 8.1.2
- `insta_assets_picker` Version: 2.3.0-dev.1

According to the [Android documentation on namespaces](https://developer.android.com/studio/build/namespace), this is a new requirement starting with AGP 8.0.

Could you please look into this issue? If it's a matter of updating the package to support the latest AGP requirements, an update would be greatly appreciated. I rely on `insta_assets_picker` for my project, and this build issue is a blocker for me.

Thank you for your work on this package and for any assistance you can provide.

Best regards,
Ayman

**Comment by LeGoffMael** [comment]
Hum, as you can see in the repository there is no android configuration in this package, so I am not sure how this issue could occurs from this package.
Did you fixed your issue ?

**Comment by aymanalhalali** [comment]
I was only able to solve it by using previous versions rather than upgrading to 8.0:

- ext.kotlin_version = '1.8.0'
- classpath 'com.android.tools.build:gradle:7.2.2'

- distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip

**Comment by AngeloAvv** [comment]
Hey @LeGoffMael , I'm having the same problem. The incriminated library should be `insta_assets_crop`, which is a direct dependency of insta_assets_picker.

```
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':insta_assets_crop'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Failed to query the value of property 'buildFlowServiceProperty'.
> Could not isolate value org.jetbrains.kotlin.gradle.plugin.statistics.BuildFlowService$Parameters_Decorated@ea90cb7 of type BuildFlowService.Parameters
   > A problem occurred configuring project ':insta_assets_crop'.
      > Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
         > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

           If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

BUILD FAILED in 780ms
Error: Gradle task assembleDebug failed with exit code 1

```

I can open an issue [here](https://github.com/LeGoffMael/image_crop) if you want

**Comment by inludese** [comment]
Simply add this to top level build.gradle into **allprojects** section
```
subprojects {
        afterEvaluate { project ->
            if (project.hasProperty('android')) {
                project.android {
                    if (namespace == null) {
                        namespace project.group
                    }
                }
            }
        }
    }
```

**Comment by AngeloAvv** [comment]
@inludese but this is not a fix, is a workaround

**Comment by LeGoffMael** [comment]
Thank you so much for your insights @AngeloAvv, I released a new `insta_assets_crop` version with the fix.
This issue should be fixed in the new `2.3.1` version !
